src/NextGen/DonationForm/resources/app/form/Form.tsx

```diff
@@ -57,16 +57,20 @@ const handleSubmitRequest = async (values, setError, gateway: Gateway) => {
             return window.frameElement.id;
         };
 
-        const {response, isRedirect} = await postData(donateUrl, {
+        const {response, isRedirectResponse} = await postData(donateUrl, {
             ...values,
             originUrl,
             isEmbed,
             embedId: getEmbedId(),
             gatewayData: beforeCreatePaymentGatewayResponse,
         });
 
-        if (isRedirect) {
-            await handleRedirect(response);
+        if (isRedirectResponse) {
+            await handleRedirect(response.url);
+        }
+
+        if (response?.type === 'redirect') {
```

Comment:
@jonwaldstein This would be a good time to use a [Type Predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates).

Reply:
@JasonTheAdams those are awesome! f95a46ed

I definitely plan on expanding these moving forward.  I think what I just added is a good start.